### PR TITLE
Disable shebang mangling for /bin/bash

### PR DESCRIPTION
**Pull request by martinky82**

### CODE REVIEW

**Comment by psss** [comment]
Thanks for the fix, @martinky82. Looking at the [packaging guidelines](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_shebang_lines) it seems there is also an option to disable the functionality for specific paths, which is recommended. Could you please update the pull request with using `%__brp_mangle_shebangs_exclude`?

Can you also use a release-notes-copy-pastable [commit message](https://tmt.readthedocs.io/en/latest/contribute.html#commits)? Thanks.

File: tmt.spec [hunk 1]
```diff
@@ -24,6 +24,9 @@ Source0: https://github.com/psss/tmt/releases/download/%{version}/tmt-%{version}
 %bcond_without englocale
 %endif
 
+# shebang mangling breaks ContOS experience
```
Comment (psss) [inline tmt.spec:27]:
```suggestion
# Shebang mangling breaks run.sh in CentOS 6 containers
```

**Review by psss — CHANGES_REQUESTED** [review]

**Comment by psss** [comment]
This does not seem to be working:

    make rpm
    ...
    rpmbuild --define '_topdir /home/psss/git/tmt/tmp' -bb tmt.spec
    error: line 28: Unknown tag: %__brp_mangle_shebangs_exclude /bin/bash
    make: *** [Makefile:47: rpm] Error 1

In order to check the shebang is as expected:

    head -1 $(rpm -ql python3-tmt|grep run.sh)

Also, could you please rebase on the latest master?

**Comment by martinky82** [comment]
It seems we're affected by this bug here:

https://bugzilla.redhat.com/show_bug.cgi?id=1839170

**Comment by psss** [comment]
Could we possibly use `%__brp_mangle_shebangs_exclude_from` to disable mangling? If I understand it correctly, this macro be used to specify paths which should be ignored. @hroncok, would you have any other recommendation?

**Comment by hroncok** [comment]
I don't understand. You build the RPM on recent Fedora or EPEL 8 and install it to CentOS 6?

**Comment by psss** [comment]
@hroncok, the issue #339 is about a `run.sh` script which is packaged in `python3-tmt` and used for executing tests in a detached way (the script is run on the remote guest to execute all tests). But when tests are executed against a CentOS 6 container the script fails, because there is no `/usr/bin/bash` in the minimal images. We want to keep `/bin/bash` in the shebang so that it works across all distros and execution methods.

**Comment by hroncok** [comment]
> the script is run on the remote guest to execute all tests

How is this achieved? Is it possible to package the script as a non-executable file and toggle the exec perm upon usage on remote guest? That would be the most clean solution.

If that is not possible, I'd set `%__brp_mangle_shebangs_exclude_from` to the script's path.
Note that it is supposed to be a regex, so better escape the dot.

**Comment by psss** [comment]
> Is it possible to package the script as a non-executable file and toggle the exec perm upon usage on remote guest? That would be the most clean solution.

Thanks for the hint. I think we could do that. The implementation should be quite straightforward.

**Comment by psss** [comment]
Here's the [relevant code](https://github.com/psss/tmt/blob/master/tmt/steps/execute/detach.py#L68) for the `detach` executor.

**Pull request closed without merging**